File: code_examples/3-Structured_Problem_Solving/12-Nested_If_Statements/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a nested if statement
 */

package main

import (
	"fmt"
)

func main() {
	// this function uses a nested if statement

	var cookiesSold int

	// input
	fmt.Print("Enter the number of boxes of cookies you sold: ")
	fmt.Scan(&cookiesSold)
	fmt.Println()

	// process & output
	if cookiesSold >= 20 {
		if cookiesSold < 30 {
			fmt.Println("You get a small prize.")
		} else {
			fmt.Println("You get a large prize.")
		}
	} else {
		fmt.Println("No prize.")
	}

	fmt.Println("\nDone.")
}

File: code_examples/3-Structured_Problem_Solving/12-Nested_If_Statements/CPP/main.cpp
// Copyright (c) 2020 St. Mother Teresa HS All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program uses a nested if statement

#include <iostream>

int main() {
    // this function uses a nested if statement
    int cookiesSold;

    // input
    std::cout << "Enter the number of boxes of cookies you sold: ";
    std::cin >> cookiesSold;
    std::cout << std::endl;

    // process & output
    if (cookiesSold >= 20) {
        if (cookiesSold < 30) {
            std::cout << "You get a small prize." << std::endl;
        } else {
            std::cout << "You get a large prize." << std::endl;
        }
    } else {
        std::cout << "No prize." << std::endl;
    }

    std::cout << "\nDone." << std::endl;
    return 0;
}
